### PR TITLE
added all function to frontend.paddle.tensor.math

### LINKED ISSUE #19391: all

### PR DESCRIPTION
**Pull request by nikita-jare**
Closes #19391

### CODE REVIEW

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi @nikita-jare, thanks a ton as usual :smile: 
- Automated Github tests have some issue, so I tried testing this locally by `pytest /workspaces/ivy/ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py::test_paddle_all` but it seems to throw a bunch of errors?
- In the test file you're editing, it looks like you aren't making use of `assume`. Could you please remove that if that's the case?
- Also, could you take care of the [lint](https://github.com/unifyai/ivy/actions/runs/5550208012/jobs/10135089510?pr=19392) issues

**Comment by nikita-jare** [comment]
Hi @rishabgit,
Please ignore the recent checkins.
Still facing some issues. I will request a review when final changes are pushed.

**Review by rishabgit — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/math.py [hunk 1]
```diff
@@ -392,3 +392,9 @@ def rsqrt(x, name=None):
 @to_ivy_arrays_and_back
 def prod(x, axis=None, keepdim=False, dtype=None, name=None):
     return ivy.prod(x, axis=axis, keepdims=keepdim, dtype=dtype)
+
+
+@to_ivy_arrays_and_back
+# @with_supported_dtypes({"2.5.0 and below": ("bool")}, "paddle")
```
[the remark this answers is not in the document]
Comment (nikita-jare) [reply ivy/functional/frontends/paddle/tensor/math.py:398]: Hi @rishabgit, for me with or without the `@with_supported_dtypes({"2.5.0 and below": ("bool")}, "paddle")`, it is running successfully. I tried now both through codespaces as well as locally.

Earlier, with `@with_supported_dtypes` uncommented, it was giving error: `KeyError:cpu`. That's why as per the suggestions from discord channel, I commented and removed it.

Now, as it is running without any errors, should I keep it commented or uncommented?

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi, tests are failing for numpy and jax, could you please look into them? Thanks! :smile: 
image

**Comment by nikita-jare** [comment]
Thank you for the review @rishabgit 
I have synced the branch with upstream, resolved the merge conflicts and ran the implementation again, all tests are passing for me when I run with codespaces. Not sure where the issue is exactly.

Screenshot 2023-08-08 at 10 55 31 AM

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hey, could you please take care of the merge conflicts? Everything lgtm to merge otherwise, thanks!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**